app/controllers/application_controller.rb

```diff
@@ -1,3 +1,12 @@
 # frozen_string_literal: true
 class ApplicationController < ActionController::Base
+  before_action :authorize
+
+  protected
+
+  def authorize
+    unless User.find_by(id: session[:user_id])
+      redirect_to(login_url, notice: "Please log in")
+    end
```

Comment:
Could shorten this up, but there's no great reason to. Strictly style choice.

```suggestion
   redirect_to(login_url, notice: "Please log in") unless User.find_by(id: session[:user_id])
```